lib/engine/phase.rb

```diff
@@ -21,6 +21,14 @@ def process_action(action)
         train = action.train
         next! if train.sym == @next_on
         rust_trains!(train, action.entity)
+        @game.companies.map do |company|
```

Comment:
this should not be map, it should be each because you have no use for the array, also make it into a method